cli/command/stack/kubernetes/ps.go

```diff
@@ -31,76 +37,73 @@ func RunPS(dockerCli *KubeCli, options options.PS) error {
 	}
 	podsClient := client.Pods()
 
-	// Fetch pods
-	if _, err := stacks.Get(namespace); err != nil {
-		return fmt.Errorf("nothing found in stack: %s", namespace)
+	filters := options.Filter.Value()
+	if err = filters.Validate(supportedPSFilters); err != nil {
+		return err
 	}
 
-	pods, err := fetchPods(namespace, podsClient)
+	// Fetch pods
+	if _, err := stacks.Get(stackName); err != nil {
+		return fmt.Errorf("nothing found in stack: %s", stackName)
+	}
+	pods, err := fetchPods(stackName, podsClient, filters)
 	if err != nil {
 		return err
 	}
 
 	if len(pods) == 0 {
-		return fmt.Errorf("nothing found in stack: %s", namespace)
+		return fmt.Errorf("nothing found in stack: %s", stackName)
```

Comment:
(just a suggestion); `fetchPods()` looks to be only used here, so this error could be returned by that function, then we can get rid of this block